frontend/src/components/CalendarView/CalendarView.jsx

```diff
@@ -122,19 +128,119 @@ function CalendarView () {
+        }
+        const backendUrlAccess = import.meta.env.VITE_BACKEND_ADDRESS;
+        const options = {
+        method: 'PUT',
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'},
+        body: JSON.stringify(updatedEvent),
+        credentials: 'include'
+          };
+        const response = await fetch(`${backendUrlAccess}/calendar/events/${info.event.id}`, options);
+        if (!response.ok) {
+          throw new Error('Something went wrong!');
+        }
+        fetchCurrentUser();
+      } catch (error) {
+        console.error('Failed to update calendar event: ', error);
+        if (error.response && error.response.status === 401 ) {
+          updateUser(null);
+        }
+        throw error;
+      }
+    }
+
+    const handleEventEditButton = async (info, id) => {
+      try {
+        const updatedEvent = {
+          title: info.title,
+          startAt: info.startAt.toISOString(),
+          endAt: info.endAt.toISOString(),
+          description: info.description,
+          location: info.location,
+          allDay: info.allDay
+        }
+        const backendUrlAccess = import.meta.env.VITE_BACKEND_ADDRESS;
+        const options = {
+        method: 'PUT',
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'},
+        body: JSON.stringify(updatedEvent),
+        credentials: 'include'
+          };
+        const response = await fetch(`${backendUrlAccess}/calendar/events/${id}`, options);
+        if (!response.ok) {
+          throw new Error('Something went wrong!');
+        }
+        fetchCurrentUser();
+        setIsModalOpen(false);
+      } catch (error) {
+        console.error('Failed to update calendar event: ', error);
+        if (error.response && error.response.status === 401 ) {
+          updateUser(null);
+        }
+        throw error;
+      }
+    }
+
+    const handleEdit = (info) => {
+      setIsDetailModalOpen(false);
+      setIsEdit(true);
+      const editInfo = {
+        id: info.id,
+        title: info.title,
+        date: getDateString(info.start),
+        start: getTimeString(info.start),
+        end: getTimeString(info.end),
+        allDay: info.allDay,
+        description: info.description,
+        location: info.location
+      }
+      setInitialView(editInfo);
+      setIsModalOpen(true);
+    }
+
+    const newEventButton = () => {
+      setInitialView({
+        title: "",
+        date: "",
+        start: "",
+        end: "",
+        allDay: false,
+        description: "",
+        location: ""
+      })
+      setIsModalOpen(true);
+      setIsEdit(false);
+    }
     return (
             <>
                 <Header />
                 <div id="event-src">
-                  <button onClick={() => setIsModalOpen(true)}>New Event</button>
+                  <button onClick={newEventButton}>New Event</button>
```

Comment:
nit: newEventButton doesn't sound like a function name. Maybe something like onNewEventClick or handleNewEventClick

Reply:
Yes, the TC1 part of editing events will be a part of a different PR, that's what I am working on now, after the code cleanup I mentioned earlier

Reply:
as for the function name, I just changed it to handleNewEventClick, thanks for the suggestion